### PR TITLE
[IMP] website_sale: Improve configurations

### PR DESCRIPTION
**Pull request by tivisse**
TaskID: 2709956

### CODE REVIEW

File: odoo/addons/base/models/res_config.py [hunk 1]
```diff
@@ -797,3 +799,9 @@ def create(self, vals_list):
                     vals.pop(field.name)
 
         return super().create(vals_list)
+
+    def action_open_template_user(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("base.action_res_users")
+        action['res_id'] = literal_eval(self.env['ir.config_parameter'].sudo().get_param('base.template_portal_user_id', 'False'))
```
Comment (Feyensv) [inline odoo/addons/base/models/res_config.py]: AFAIK, it's supposed to be an int, is literal_eval really necessary here ? :thinking:

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

File: addons/crm/models/res_config_settings.py [hunk 2]
```diff
@@ -130,23 +130,32 @@ def _compute_predictive_lead_scoring_field_labels(self):
                 setting.predictive_lead_scoring_field_labels = _('Stage')
 
     def set_values(self):
-        group_lead_before = self.env.ref('crm.group_use_lead') in self.env.user.groups_id
+        group_use_lead = self.env.ref('crm.group_use_lead')
```
Comment (tde-banana-odoo) [inline addons/crm/models/res_config_settings.py:133]: Actually you could xmlid_to_resid directly to further gain queries. This group is a master data anyway.

File: addons/website_event_jitsi/views/res_config_settings_views.xml [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
     <record id="res_config_settings_view_form" model="ir.ui.view">
-        <field name="name">res.config.settings.view.form.inherit.website.jitsi</field>
+        <field name="name">res.config.settings.view.form.inherit.event</field>
```
Comment (tde-banana-odoo) [inline addons/website_event_jitsi/views/res_config_settings_views.xml:4]: Guidelines: [...] inherit.website.event.jitsi

File: addons/website_event_jitsi/views/res_config_settings_views.xml [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
         <field name="model">res.config.settings</field>
-        <field name="inherit_id" ref="website.res_config_settings_view_form" />
+        <field name="inherit_id" ref="event.res_config_settings_view_form" />
         <field name="arch" type="xml">
-            <xpath expr="//div[@id='google_maps_setting']" position="after">
+            <xpath expr="//div[@name='events_setting_container']" position="inside">
```
Comment (tde-banana-odoo) [inline addons/website_event_jitsi/views/res_config_settings_views.xml:8]: Using inside leads to a somehow undefined ordering depending on module installation order. Doesn't that break anything ? Could be solved by using priorities.

File: addons/website_event_track/models/website.py [hunk 5]
```diff
@@ -14,8 +13,18 @@
 class Website(models.Model):
     _inherit = "website"
 
-    app_icon = fields.Image(string='Website App Icon', compute='_compute_app_icon', store=True, readonly=True, help='This field holds the image used as mobile app icon on the website (PNG format).')
-    events_app_name = fields.Char(string='Events App Name', compute='_compute_events_app_name', store=True, readonly=False, help="This fields holds the Event's Progressive Web App name.")
+    app_icon = fields.Image(
```
Comment (tde-banana-odoo) [inline addons/website_event_track/models/website.py:16]: Unnecessary diff, creates issues and is takes too much space for nothing( moreover we cannot see what changed, is there any difference ?)

File: odoo/addons/base/models/res_config.py (hunk 1 above)
Comment (tde-banana-odoo) [reply odoo/addons/base/models/res_config.py:805]: And do we intend to land on a void form view if the config parameter is not set ? Shouldn't we check it exists (as it is not synchronized, user could have been removed) ?

File: addons/auth_signup/models/res_config_settings.py [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
-    ], string='Customer Account', default='b2b', config_parameter='auth_signup.invitation_scope')
-    auth_signup_template_user_id = fields.Many2one('res.users', string='Template user for new users created through signup',
-                                                   config_parameter='base.template_portal_user_id')
-
-    def open_template_user(self):
-        action = self.env["ir.actions.actions"]._for_xml_id("base.action_res_users")
-        action['res_id'] = literal_eval(self.env['ir.config_parameter'].sudo().get_param('base.template_portal_user_id', 'False'))
-        action['views'] = [[self.env.ref('base.view_users_form').id, 'form']]
-        return action
+    auth_signup_reset_password = fields.Boolean(
```
Comment (tde-banana-odoo) [inline addons/auth_signup/models/res_config_settings.py:10]: Lot of unnecessary diff in this file, makes real changes hard to understand. Could you maybe limit to real changes ?

File: addons/crm/models/res_config_settings.py [hunk 7]
```diff
@@ -130,23 +130,32 @@ def _compute_predictive_lead_scoring_field_labels(self):
...
+                'active': self.crm_use_auto_assignment and self.crm_auto_assignment_action == 'auto',
+                'interval_type': self.crm_auto_assignment_interval_type,
+                'interval_number': self.crm_auto_assignment_interval_number,
+                # keep nextcall on cron as it is required whatever the setting
+                'nextcall': self.crm_auto_assignment_run_datetime if self.crm_auto_assignment_run_datetime else assign_cron.nextcall,
+            }
+            cron_vals = {field_name: value for field_name, value in cron_vals.items() if assign_cron[field_name] != value}
+            if cron_vals:
+                assign_cron.write(cron_vals)
         # TDE FIXME: re create cron if not found ?
```
Comment (tde-banana-odoo) [inline addons/crm/models/res_config_settings.py:159]: My fixme is still there, ouf :D .

File: addons/hr_expense/models/res_config_settings.py [hunk 8]
```diff
@@ -6,8 +6,11 @@
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    expense_alias_prefix = fields.Char('Default Alias Name for Expenses', compute='_compute_expense_alias_prefix',
-        store=True, readonly=False)
+    expense_alias_prefix = fields.Char(
```
Comment (tde-banana-odoo) [inline addons/hr_expense/models/res_config_settings.py:9]: Unnecessary diff ?

File: addons/website/models/res_config_settings.py [hunk 9]
```diff
@@ -14,116 +13,173 @@ class ResConfigSettings(models.TransientModel):
...
-    has_social_network = fields.Boolean("Configure Social Network", compute=has_social_network, inverse=inverse_has_social_network)
-
-    favicon = fields.Binary('Favicon', related='website_id.favicon', readonly=False)
-    social_default_image = fields.Binary('Default Social Share Image', related='website_id.social_default_image', readonly=False)
-
-    google_maps_api_key = fields.Char(related='website_id.google_maps_api_key', readonly=False)
-    group_multi_website = fields.Boolean("Multi-website", implied_group="website.group_multi_website")
-
-    @api.onchange('website_id')
+    website_id = fields.Many2one(
```
Comment (tde-banana-odoo) [inline addons/website/models/res_config_settings.py:16]: Waaaaay too much diff that breaks forward ports, history,...

**Review by tde-banana-odoo — COMMENTED** [review]
Validation for SM part: some small comments.

Apart from that, there is a lot of noisy changes (unnecessary line returns, ...). This would break history, forward ports, and prevents from reading actual changes. If possible could you revert those ?

**Comment by tde-banana-odoo** [comment]
Small note for twitter email: if I understand it, purpose would be to have admin emial as default paypal account email. It will not work on our saas I think as it does not go through -i of modules. This means this feature will probably not work on our saas, and added value in local / onsite is very low, as chances it helps you setting your paypal account are about 0.

If this is wanted this should be done internal side, like they did for mass mailing.

Cheers !

File: addons/website_event_jitsi/views/res_config_settings_views.xml (hunk 4 above)
Comment (tivisse) [reply addons/website_event_jitsi/views/res_config_settings_views.xml:8]: In this case, there's only one override.

But even if there were many of them, this is not a big deal in our case. The goal is to group the settings by thematic, and there are generally between 1 to 8 settings by section.

**Comment by tivisse** [comment]
@tde-banana-odoo Concerning this: https://github.com/odoo/odoo/pull/82999#discussion_r796710002

It's just a duplicated code that I moved to ``base`` module, but I'll adapt it. 

About your comment, just for your information:
- The config parameter is always set: see https://github.com/odoo/odoo/blame/master/odoo/addons/base/security/base_groups.xml#L89
- If the parameter of the template user is removed, you'll have bigger issues as the user creation will raise.

File: addons/auth_signup/models/res_config_settings.py (hunk 6 above)
Comment (tivisse) [reply addons/auth_signup/models/res_config_settings.py:10]: Well, those files are often messy and chunk of codes are copy/pasted everywhere because people seldom understand how it is supposed to work. 

As I made a round on all the files, I took the opportunity to make them clean. 

Concerning the 1 line by attribute, this is quite subjective but it allows to limit the modifications of 1 attribute to 1 line (and not the whole one) to those fields that have more field attributes in general than in common odoo models, and actually limit the forward port conflicts (once you passed this commit, I agree with you).

This will be quite time-consuming to revert only those changes, so I'll keep it that way.

File: addons/payment_paypal/data/payment_acquirer_data.xml [hunk 10]
```diff
@@ -8,6 +8,7 @@
         <field name="support_fees_computation">True</field>
         <field name="support_refund"></field>
         <field name="support_tokenization">False</field>
+        <field name="paypal_email_account" model="res.users" eval="obj().env.ref('base.user_admin').email or ''"/>
```
[the remark this answers is not in the document]
Comment (tivisse) [reply addons/payment_paypal/data/payment_acquirer_data.xml:11]: This is the email that is used to communicate with paypal, email address that will be used by them to contact you personally to onboard you during your account creation.

I understand why it won't work on SaaS instances, I didn't think about that, thanks :)

I'll revert this change, and suggest to make an internal custo.

File: addons/website/models/res_config_settings.py (hunk 9 above)
Comment (tivisse) [reply addons/website/models/res_config_settings.py:16]: I agree for forward ports, but res.config fixes are not done on a daily basis too, and are most likely to start from this version.

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff
